graphql/src/store/query.rs

```diff
@@ -141,6 +141,24 @@ fn build_fulltext_filter_from_object(
     )
 }
 
+fn parse_change_block_filter(
+    value: &graph::data::value::Value,
+) -> Result<BlockNumber, QueryExecutionError> {
+    match value {
+        graph::data::value::Value::Object(object) => {
+            let block_value = object
+                .get("number_gte")
+                .ok_or_else(|| QueryExecutionError::InvalidFilterError)?;
+
+            match block_value {
+                graph::data::value::Value::Int(n) => Ok(*n as i32),
```

Comment:
There's a helper for this conversion: use `i32::try_from_value(block_value).map_err(|_| QueryExecutionError::InvalidFilterError)` instead of the `match` statement to do a conversion that checks bounds.

Reply:
agree, will refactor :)